sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -837,6 +837,48 @@ def tx_disable_channel(self, channel, disable):
 
         return self.xcvr_eeprom.write(consts.TX_DISABLE_FIELD, channel_state)
 
+    def get_rx_disable_support(self):
+	return not self.is_flat_memory() and self.xcvr_eeprom.read(consts.RX_DISABLE_SUPPORT_FIELD)
+
+    def get_rx_disable(self):
+	rx_disable_support = self.get_rx_disable_support()
+	if rx_disable_support is None:
+	    return None
+	if not rx_disable_support:
+	    return ["N/A" for _ in range(self.NUM_CHANNELS)]
+	rx_disable = self.xcvr_eeprom.read(consts.RX_DISABLE_FIELD)
+	if rx_disable is None:
+	    return None
+	return [bool(rx_disable & (1 << i)) for i in range(self.NUM_CHANNELS)]
+
+    def rx_disable(self, rx_disable):
+	val = 0xFF if rx_disable else 0x0
+	return self.xcvr_eeprom.write(consts.RX_DISABLE_FIELD, val)
```

Comment:
@vdahiya12 Wondering if we can call rx_disable_channel function instead here and pass the `channel` as 0xff

Reply:
here i used the same way we implement tx_disable for code coherency